prediction_market_agent/db/pinecone_handler.py

```diff
@@ -77,36 +108,38 @@ def insert_all_omen_markets_if_not_exists(
             sort_by=SortBy.NEWEST,
             created_after=created_after,
         )
-        texts = [m.question_title for m in markets]
-        metadatas = [
-            PineconeMetadata.from_omen_market(market).model_dump() for market in markets
-        ]
+        missing_markets = self.filter_markets_already_in_index(markets=markets)
+
+        texts = []
+        metadatas = []
+        for m in missing_markets:
+            texts.append(m.question_title)
+            metadatas.append(PineconeMetadata.from_omen_market(m).model_dump())
+
         if texts:
             n_elements = 100
             chunked_texts = list(self.chunks(texts, n_elements))
             chunked_metadatas = list(self.chunks(metadatas, n_elements))
             for text_chunk, metadata_chunk in tqdm(
                 zip(chunked_texts, chunked_metadatas), total=len(chunked_texts)
             ):
+                ids_chunk = [self.encode_text(text) for text in text_chunk]
                 logger.debug(f"Inserting {len(text_chunk)} into the vector database.")
-                self.insert_texts_if_not_exists(
-                    texts=text_chunk, metadatas=metadata_chunk
+                self.insert_texts(
+                    ids=ids_chunk, texts=text_chunk, metadatas=metadata_chunk
                 )
 
     def find_nearest_questions_with_threshold(
-        self, limit: int, text: str, threshold: float = 0.7
+        self, limit: int, text: str, threshold: float = 0.25
     ) -> list[PineconeMetadata]:
         # Note that pagination is not implemented in the Pinecone client.
         # Hence we set a large limit and hope we get enough results that satisfy the threshold.
-        documents_and_scores = self.vectorstore.similarity_search_with_score(
-            query=text, k=int(limit * 5)
+        documents_and_scores = self.vectorstore.similarity_search_with_relevance_scores(
+            query=text, k=limit, score_threshold=threshold
         )
```

Comment:
`similarity_search_with_relevance_scores` is doing the trick for finding more relevant markets, together with the larger embedding